hadoop-jobs/src/test/java/com/linkedin/hadoop/jobs/HdfsWaitJobTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.linkedin.hadoop.jobs;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+
+import org.apache.log4j.Logger;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Hadoop mini-cluster and regular unit testing for HdfsWaitJob.java.
+ */
+public class HdfsWaitJobTest {
+
+  private static final Logger _log = Logger.getLogger(HdfsWaitJobTest.class);
+  private Configuration _conf;
+  private int _dataNodes;
+  private MiniDFSCluster _dfsCluster;
+  private FileSystem _dfsFileSystem;
+
+  /**
+   * Constructor for the HdfsWaitJobTest class that initializes the mini-cluster
+   * with one HDFS data node.
+   */
+  public HdfsWaitJobTest() {
+    _dataNodes = 1;
+  }
+
+  /**
+   * Shuts down the mini-cluster after the unit tests are finished.
+   *
+   * @throws Exception If there is a problem shutting down the mini-cluster
+   */
+  @AfterClass
+  public void afterClass() throws Exception {
+    if (_dfsCluster != null) {
+      _log.info("*** Shutting down Mini-DFS Cluster");
+      _dfsCluster.shutdown();
+      _dfsCluster = null;
+    }
+  }
+
+  /**
+   * Initializes the mini-cluster to use with unit tests.
+   *
+   * @throws Exception If there is a problem initializing the mini-cluster
+   */
+  @BeforeClass
+  public void beforeClass() throws Exception {
+    _conf = new Configuration();
+    _conf.set("mapreduce.framework.name", "local");
+ 
+    _log.info("*** Starting Mini DFS Cluster");
+    _dfsCluster = new MiniDFSCluster(_conf, _dataNodes, true, null);
```

Comment:
Use the MiniDFSCluster.Builder method shown in https://github.com/linkedin/linkedin-gradle-plugin-for-apache-hadoop/blob/master/example-project/src/test/java/com/linkedin/hadoop/example/minicluster/MapReduceClusterTestBase.java